### ISSUE TITLE
Runtime error on develop branch with DWF invert

### ISSUE THREAD

**Issue by cpviolator** [lattice/quda#749]
I'm hooking up QLUA with QUDA's DWF inverter. Compiling the latest QUDA develop and running wilson/clover inverter are fine. The invert test for a mobius domain wall action gives the following runtime error.

```
[howarth@scc-k02 tests]$ ./invert_test --sdim 16 --tdim 16 --dslash-type mobius --Lsdim 6
Enabling peer-to-peer copy engine and direct load/store access
Disabling GPU-Direct RDMA access
Rank order is column major (t running fastest)
running the following test:
prec    prec_sloppy   multishift  matpc_type  recon  recon_sloppy S_dimension T_dimension Ls_dimension   dslash_type  normalization
single   single          0        even-even     18     18          16/ 16/ 16      16          6               mobius     kappa
Grid partition info:     X  Y  Z  T
                         0  0  0  0
QUDA 0.9.1 (git v0.9.0-703-g419f7b0-dirty-sm_60)
CUDA Driver version = 9020
CUDA Runtime version = 8000
Found device 0: Tesla P100-PCIE-16GB
Using device 0: Tesla P100-PCIE-16GB
WARNING: Data reordering done on GPU (set with QUDA_REORDER_LOCATION=GPU/CPU)
WARNING: Using device memory pool allocator
WARNING: Using pinned memory pool allocator
Loaded 32 sets of cached parameters from /projectnb/isingmag/howarth/QLUA/quda_build/tests/tunecache.tsv
Computed plaquette is 1.223368e-01 (spatial = 1.224580e-01, temporal = 1.222156e-01)
Tuned block=(32,1,1), grid=(1,1,1), shared=0,  giving 0.00 Gflop/s, 8.62 GB/s for cudaMemcpyHostToDevice with loadSpinorField,cuda_color_spinor_field.cu,599
Tuned block=(64,1,1), shared=0,  giving 0.00 Gflop/s, 492.89 GB/s for N4quda15CopyColorSpinorIfdLi4ELi3ENS_18CopyColorSpinorArgINS_11colorspinor11FloatNOrderIfLi4ELi3ELi4ELb0EEENS2_21SpaceSpinorColorOrderIdLi4ELi3EEEEEEE with out_stride=196608,in_stride=196608,NonRelBasis
Tuned block=(480,1,1), grid=(108,1,1), shared=0,  giving 211.20 Gflop/s, 422.41 GB/s for N4quda4blas5Norm2Id6float26float4EE with vol=196608,stride=196608,precision=4,Ns=4,Nc=3
Source: CPU = 1.57173e+06, CUDA copy = 1.57173e+06
Solution: CPU = 0, CUDA copy = 0
Prepared source = 1.57173e+06
Prepared solution = 0
Prepared source post mass rescale = 1.57173e+06
Creating a CG solver
Tuned block=(32,1,1), grid=(1,1,1), shared=0,  giving 0.00 Gflop/s, 494.61 GB/s for cudaMemcpyAsyncDeviceToDevice with copy,copy_quda.cu,133
CG: 0 iterations, <r,r> = 1.571728e+06, |r|/|b| = 1.000000e+00
ERROR: Requested block size 544x2x1=1088 greater than hardware limit 1024 (rank 0, host scc-k02, /projectnb/isingmag/howarth/QLUA/quda/include/tune_quda.h:281 in checkLaunchParam())
       last kernel called was (name=N4quda16MDWFDslashPCCudaI6float4S1_EE,volume=8x16x16x16x6,aux=policy_kernel=interior,ghost=0000,comm=0000,reconstruct=18,Dslash4pre)
Saving 43 sets of cached parameters to /projectnb/isingmag/howarth/QLUA/quda_build/tests/tunecache_error.tsv
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
```

I added a little extra to the output which shows that `param.block.x = 544`. This looks like a tuning error. Any ideas? Here's my environment:

```Currently Loaded Modulefiles:
  1) gcc/4.8.1                4) gsl/2.3                  7) openmpi/2.1.0_gcc4.8.1  10) openblas/0.2.20         13) gmp/6.1.2               16) fftw/3.3.6
  2) cuda/8.0                 5) libcurl/7.48.0           8) emacs/25.3              11) lapack/3.8.0            14) mpfr/4.0.1              17) autoconf/2.69
  3) eigen/3.3.4              6) git/2.6.3                9) cmake/3.12.3            12) pgi/18.4                15) hdf5/1.8.18_gcc-6.2.0   18) automake/1.15```

**Comment by cpviolator** [comment]
Possible fix?

Replace line 106:

```if ( advance[0] || advance[1]) {```
 
goes to 

```if ( (advance[0] || advance[1]) && param.block.x*param.block.y*param.block.z+1 < deviceProp.maxThreadsPerBlock) {```

The mobius inverter runs smoothly, but I'm not sure if this makes the tuner inefficient in some way. Also, this may have to be propagated to other DWF actions.

**Comment by cpviolator** [comment]
Fixed with #750
